bouter/decorators.py

```diff
@@ -0,0 +1,60 @@
+import warnings
+import functools
+import flammkuchen as fl
+
+from numpy import VisibleDeprecationWarning
+
+
+CACHE_FILE_TEMPLATE = "cache_{}.h5"
+
+
+def cache_results(method):
+    """ Method decorator that caches an .h5 file with the results of the
+    decorated function. This behavior can be disabled with the exp.cache_active
+    flag.
+    Function results are loaded if the new call arguments match the old call,
+    or if the exp.default_cached flag is set.
+    :param method:
+    :return:
+    """
+
+    def decorated_method(exp, **kwargs):
+        if exp.cache_active:
+            # Create cache file if none exists:
+            filename = exp.root / CACHE_FILE_TEMPLATE.format(method.__name__)
```

Comment:
one experiment can have multiple sessions in the same folder, should we not rather prefix with session_id?

Reply:
good point, will add that